doc/source/development/rfc/rfc100_float16_support.rst

```diff
@@ -0,0 +1,142 @@
+.. _rfc-100:
+
+=============================
+RFC 100: Support float16 type
+=============================
+
+============== =============================================
+Author:        Erik Schnetter
+Contact:       eschnetter @ perimeterinstitute.ca
+Started:       2024-Jun-05
+Status:        Proposed
+Target:        GDAL 3.11
+============== =============================================
+
+Summary
+-------
+
+This RFC adds support for the IEEE 16-bit floating point data type
+(aka ``half``, ``float16``). It adds new pixel data types
+``GDT_Float16`` and ``GDT_CFloat16``, backed by a new ``GFloat16`` C++
+type. This type will be emulated in software if not available
+natively.
+
+Some drivers, in particular Zarr, will be extended to support this
+type. Drivers that do not support this type will be checked to ensure
+they report an appropriate errors when necessary.
```

Comment:
`an appropriate errors` - mismatch between `an` and `s` on error

Reply:
Done.